src/libraries/lrt/RenzoLibrary.sol

```diff
@@ -227,28 +198,28 @@ library RenzoLibrary {
         //        uint256 mintAmount = newEzETHSupply - _existingEzETHSupply;
         //
         // Solve for newEzETHSupply
-        uint256 newEzEthSupply = (amountOut + totalSupply);
-        uint256 newEzEthSupplyRay = newEzEthSupply.scaleUpToRay(18);
+        uint256 newEzEthSupply = (amountOut + _existingEzETHSupply);
 
         //        uint256 newEzETHSupply = (_existingEzETHSupply * SCALE_FACTOR) / (SCALE_FACTOR -
         // inflationPercentage);
         //
         // Solve for inflationPercentage
-        uint256 intem = totalSupply.scaleUpToRay(18).mulDiv(RAY, newEzEthSupplyRay);
-        uint256 inflationPercentage = RAY - intem;
+        uint256 inflationPercentage = WAD - WAD.mulDiv(_existingEzETHSupply, newEzEthSupply);
 
         //         uint256 inflationPercentage = SCALE_FACTOR * _newValueAdded / (_currentValueInProtocol +
         // _newValueAdded);
         //
         // Solve for _newValueAdded
-        uint256 ethAmountRay = inflationPercentage.mulDiv(totalTVL.scaleUpToRay(18), RAY - inflationPercentage);
+        uint256 ethAmountIn = inflationPercentage.mulDiv(_currentValueInProtocol, WAD - inflationPercentage);
 
-        // Truncate from RAY to WAD with roundingUp plus one extra
-        // The one extra to get into the next mint range
-        uint256 ethAmount = ethAmountRay / 1e9 + 1;
-        if (ethAmountRay % 1e9 != 0) ++ethAmount;
+        if (inflationPercentage * _currentValueInProtocol % (WAD - inflationPercentage) != 0) {
+            // Unlikely to overflow
+            unchecked {
+                ethAmountIn++;
+            }
+        }
```

Comment:
What's the reason for manually implementing rounding up instead of just using `mulDiv` with rounding up.

Reply:
No specific reason. Changed to use mulDiv round up.